cellmlmanip/units.py

```diff
@@ -368,13 +368,34 @@ def get_conversion_factor(self,
 
         if from_unit is not None:
             assert isinstance(from_unit, self.ureg.Unit), 'from_unit must be of type pint:Unit'
-            return self.convert_to(1 * from_unit, to_unit).magnitude
+            conversion_factor = self.convert_to(1 * from_unit, to_unit).magnitude
         elif quantity is not None:
             assert isinstance(quantity, self.ureg.Quantity), 'quantity must be of type pint:Quantity'
-            return self.convert_to(quantity, to_unit).magnitude
+            conversion_factor = self.convert_to(quantity, to_unit).magnitude
         else:
             assert isinstance(expression, sympy.Expr), 'expression must be of type Sympy expression'
-            return self.convert_to(1 * self.summarise_units(expression), to_unit).magnitude
+            conversion_factor = self.convert_to(1 * self.summarise_units(expression), to_unit).magnitude
+
+        if math.isclose(conversion_factor, 1.0):
+            return 1.0
+        else:
+            return conversion_factor
+
+    def equivalent_units(self, unit_or_expr1, unit_or_expr2):
```

Comment:
Is this the same as `is_unit_equal` ?

Reply:
Actually yes I didn't notice, let me remove my new one.